src/site/antora/modules/ROOT/pages/manual/pattern-layout.adoc

```diff
@@ -0,0 +1,1719 @@
+
+Replaces occurrences of a regular expression (`regex`) with its replacement (`substitution`) in the string resulting from the evaluation of the pattern
+
+.link:../javadoc/log4j-core/org/apache/logging/log4j/core/pattern/RegexReplacementConverter.html[`RegexReplacementConverter`] specifier grammar
+[source,text]
+----
+replace{pattern}{regex}{substitution}
+----
+
+For example, `%replace{%msg}{\s}{}` will remove all spaces contained in the event message.
+
+The pattern can be arbitrarily complex and in particular, can contain multiple conversion keywords.
+For instance, `%replace{%logger %msg}{\.}{/}` will replace all dots in the logger or the message of the event with a forward slash.
+
+[#converter-rootException]
+==== Root exception
+
+The same as xref:#converter-exception[the `exception` converter], but the stack trace is printed starting with the first exception in the causal chain that was thrown followed by each subsequent wrapping exception
+
+.link:../javadoc/log4j-core/org/apache/logging/log4j/core/pattern/RootThrowablePatternConverter.html[`RootThrowablePatternConverter`] specifier grammar
+[source,text]
+----
+rEx|rException|rThrowable
+  {
+    ["none" | "short" | "full" | depth]
+    [,filters(package,package,...)]
+    [,separator(separator)]
+  }
+  {ansi(
+    Key=Value,Value,...
+    Key=Value,Value,...
+    ...)
+  }
+  {suffix(pattern)}
+----
+
+The throwable conversion specifier can be followed by an option in the form `%rEx\{short}`, which will only output the first line of the `Throwable`, or `%rEx\{n}`, where the first `n` lines of the stack trace will be printed.
+
+Specifying `%rEx\{none}` or `%rEx\{0}` will suppress printing of the exception.
+
+Use `filters(packages)`, where `packages` is a list of package names to suppress matching stack frames from stack traces.
+
+Use a `separator` string to separate the lines of a stack trace, e.g., `separator(|)`.
+The default value is the `line.separator` system property, which is platform dependent.
+
+Use `rEx{suffix(pattern)}` to add the output of `pattern` to the output only when there is a `Throwable` to print.
+
+[#converter-seq]
+==== Sequence number
+
+Includes a sequence number that will be incremented in every event
+
+.link:../javadoc/log4j-core/org/apache/logging/log4j/core/pattern/SequenceNumberPatternConverter.html[`SequenceNumberPatternConverter`] specifier grammar
+[source,text]
+----
+sn
+sequenceNumber
+----
+
+The counter is a static variable, so will only be unique within applications that share the same converter class object.
+
+[#converter-style]
+==== Style
+
+Use ANSI escape sequences to style the result of the enclosed pattern.
+Windows users should refer to xref:#jansi[].
+
+.link:../javadoc/log4j-core/org/apache/logging/log4j/core/pattern/StyleConverter.html[`StyleConverter`] specifier grammar
+[source,text]
+----
+style{pattern}{ANSI style}
+----
+
+The style can consist of a comma-separated list of style names from the following table:
```

Comment:
For compatibility with `%highlight` this can also be a space-separated list.

Reply:
I'd appreciate it if you can address this while improving the section on `%highlight`.